src/main/java/ua/su/service/impl/ClinicServiceImpl.java

```diff
@@ -9,13 +9,12 @@
 import ua.su.service.ClinicService;
 
 import java.util.List;
-import java.util.stream.Collectors;
 
 @Service
```

Comment:
You can remove @Service here and use more specific implementation.

Reply:
I removed @service from ClinicServiceImpl.java and added to the child classes.